src/zenml/integrations/wandb/experiment_trackers/wandb_experiment_tracker.py

```diff
@@ -131,25 +127,50 @@ def cleanup_step_run(self, info: "StepRunInfo", step_failed: bool) -> None:
 
     def _initialize_wandb(
         self,
+        info: "StepRunInfo",
         run_name: str,
         tags: List[str],
-        settings: Union["Settings", Dict[str, Any], None] = None,
     ) -> None:
         """Initializes a wandb run.
 
         Args:
+            info: Step run information.
             run_name: Name of the wandb run to create.
             tags: Tags to attach to the wandb run.
-            settings: Additional settings for the wandb run.
         """
         logger.info(
             f"Initializing wandb with entity {self.config.entity}, project "
             f"name: {self.config.project_name}, run_name: {run_name}."
         )
+        settings = cast(
```

Comment:
This returns a non-optional, so you can get rid of the `if settings and ...` part in those conditions.